flatdata-py/tests/test_archive_creation.py

```diff
@@ -0,0 +1,237 @@
+from flatdata.generator.engine import Engine
+#from flatdata.lib.archive_builder import FileResourceWriter
+from common import DictResourceStorage, INSTANCE_TEST_SCHEMA, RESOURCE_PAYLOAD, ARCHIVE_SIGNATURE_PAYLOAD
+
+from flatdata.lib.resource_storage import _Resource
+
+
+class DummyResourceWriter:
+    """
+    Mimick `FileResourceWriter` and store all that is written in the dict `data`.
+    """
+    def __init__(self):
+        self.data = dict()
+
+    def get(self, key, is_subarchive=False):   
+        if is_subarchive:
+            return DummyResourceWriter()
+        
+        if key not in self.data:
+            self.data[key] = bytearray()
+
+        self.data[key] = DummyFileWriter(key, self.data[key])
+        return self.data[key]
+    
+    def open(self, name, file_path):
+        pass
+
+    def write(self, data):
+        print("#1 write called")
+        pass
+
+    def close(self):
+        pass
+
+class DummyFileWriter(_Resource):
+    """
+    Mimick binary file writing and store result in `data` of
+    the `BytesResourceWriter` that created it.
+    """
+    def __init__(self, key, data):
+        super().__init__(key, path="dummypath")
+
+def test_create_example_archive():
+    module = Engine(INSTANCE_TEST_SCHEMA).render_python_module()
+    memwrite = DummyResourceWriter()
+
+    builder = module.backward_compatibility_ArchiveBuilder(memwrite)
+    builder.set("resource", {"a": -0x1, "b": 0x01234567, "c": -0x28, "d": 0})
+    builder.finish()
+
+    valid_data = {
+        "Archive.archive": ARCHIVE_SIGNATURE_PAYLOAD,
+        "Archive.archive.schema": module.backward_compatibility_Archive.schema().encode(),
+        "resource": RESOURCE_PAYLOAD,
+        "resource.schema": module.backward_compatibility_Archive.resource_schema('resource').encode()
+    }
+
+    for (vkey, vdata) in valid_data.items():
+        assert memwrite.data[vkey].get_data() == vdata, f'"{vkey}" is "{memwrite.data[vkey]}", should be "{vdata}"'
+
+def test_create_vector_archive():
+    vector_test_schema = """
+namespace backward_compatibility {
+    struct SignedStruct {
+        a : i16 : 5;
+        b : u32 : 32;
+        c : i32 : 7;
+        d : u32 : 32;
+    }
+    archive Archive {
+        resource: vector< SignedStruct >;
+    }
+}
+"""
+    RESOURCE_PAYLOAD = (
+        b"\x14\x00\x00\x00\x00\x00\x00\x00"  # Payload size in bytes
+        b"\xff\xac\x68\x24\x00\x0b\x00\x00"  # Payload
+        b"\x00\x00\xff\xac\x68\x24\x00\x0b"  # Payload
+        b"\x00\x00\x00\x00"  # Payload
+        b"\x00\x00\x00\x00\x00\x00\x00\x00"  # Padding
+    )
```

Comment:
Since multiple tests are using the same data (tests for reading use it as well), does it make sense to de-duplicate it?